src/renderers/shared/reconciler/__tests__/ReactCompositeComponent-test.js

```diff
@@ -874,6 +874,105 @@ describe('ReactCompositeComponent', function() {
     expect(div.children[0].id).toBe('aliens');
   });
 
+  it('should trigger componentWillReceiveProps for context changes', function() {
+    var grandChildCount = 0;
+    var childCount = 0;
+
+    var GrandChild = React.createClass({
+      contextTypes: {
+        foo: ReactPropTypes.string.isRequired,
+      },
+
+      componentWillReceiveProps: function(nextProps, nextContext) {
+        expect('foo' in nextContext).toBe(true);
+        grandChildCount++;
+      },
+
+      render: function() {
+        return <span className="grand-child">{this.props.children}</span>;
+      },
+    });
+
+    var ChildWithContext = React.createClass({
+      contextTypes: {
+        foo: ReactPropTypes.string.isRequired,
+      },
+
+      componentWillReceiveProps: function(nextProps, nextContext) {
+        expect('foo' in nextContext).toBe(true);
+        childCount++;
+      },
+
+      render: function() {
+        return <div className="child-with">{this.props.children}</div>;
+      },
+    });
+
+    var ChildWithoutContext = React.createClass({
+      componentWillReceiveProps: function(nextProps, nextContext) {
+        expect('foo' in nextContext).toBe(false);
+        childCount++;
+      },
+
+      render: function() {
+        return <div className="child-without">{this.props.children}</div>;
+      },
+    });
+
+    var Parent = React.createClass({
+      childContextTypes: {
+        foo: ReactPropTypes.string,
+      },
+
+      getChildContext: function() {
+        return {
+          foo: this.props.cntxt,
+        };
+      },
+
+      render: function() {
+        return <div className="parent">{this.props.children}</div>;
+      },
+    });
+
+    Parent.ChildWithContext = ChildWithContext;
+    Parent.ChildWithoutContext = ChildWithoutContext;
+    Parent.GrandChild = GrandChild;
+
+    var div = document.createElement('div');
+
+    ReactDOM.render(<Parent cntxt="foo">
+      <Parent.ChildWithoutContext>
```

Comment:
This is a little weird.  Why are you using `Parent.ChildWithoutContext` instead of `ChildWithoutContext`?